tests/testsuite/profile_custom.rs

```diff
@@ -0,0 +1,319 @@
+        "#,
+        )
+        .file("src/lib.rs", "")
+        .file("xxx/Cargo.toml", &basic_lib_manifest("xxx"))
+        .file("xxx/src/lib.rs", "")
+        .file("yyy/Cargo.toml", &basic_lib_manifest("yyy"))
+        .file("yyy/src/lib.rs", "")
+        .build();
+
+    // profile overrides are inherited between profiles using inherits and have a
+    // higher priority than profile options provided by custom profiles
+    p.cargo("build -v")
+        .masquerade_as_nightly_cargo()
+        .with_stderr_unordered(
+            "\
+[COMPILING] xxx [..]
+[COMPILING] yyy [..]
+[COMPILING] foo [..]
+[RUNNING] `rustc --crate-name xxx [..] -C codegen-units=5 [..]`
+[RUNNING] `rustc --crate-name yyy [..] -C codegen-units=3 [..]`
+[RUNNING] `rustc --crate-name foo [..] -C codegen-units=7 [..]`
+[FINISHED] dev [unoptimized + debuginfo] target(s) in [..]
+",
+        )
+        .run();
+
+    // This also verifies that the custom profile names appears in the finished line.
+    p.cargo("build --profile=other -Z unstable-options -v")
+        .masquerade_as_nightly_cargo()
+        .with_stderr_unordered(
+            "\
+[COMPILING] xxx [..]
+[COMPILING] yyy [..]
+[COMPILING] foo [..]
+[RUNNING] `rustc --crate-name xxx [..] -C codegen-units=5 [..]`
+[RUNNING] `rustc --crate-name yyy [..] -C codegen-units=6 [..]`
+[RUNNING] `rustc --crate-name foo [..] -C codegen-units=2 [..]`
+[FINISHED] other [unoptimized + debuginfo] target(s) in [..]
+",
+        )
+        .run();
+}
+
+#[cargo_test]
+fn conflicting_usage() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [package]
+            name = "foo"
+            version = "0.0.1"
+            authors = []
+        "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("build -Z unstable-options --profile=dev --release")
+        .masquerade_as_nightly_cargo()
+        .with_status(101)
+        .with_stderr_unordered("error: Conflicting usage of --profile and --release")
+        .run();
+
+    p.cargo("install -Z unstable-options --profile=release --debug")
+        .masquerade_as_nightly_cargo()
+        .with_status(101)
+        .with_stderr_unordered("error: Conflicting usage of --profile and --debug")
+        .run();
+}
+
+#[cargo_test]
+fn clean_custom_dirname() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            cargo-features = ["named-profiles"]
+
+            [package]
+            name = "foo"
+            version = "0.0.1"
+            authors = []
+
+            [profile.other]
+            inherits = "release"
+        "#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .build();
+
+    p.cargo("build --release")
+        .masquerade_as_nightly_cargo()
+        .run();
+
+    p.cargo("clean -p foo").masquerade_as_nightly_cargo().run();
+
+    p.cargo("build --release")
+        .masquerade_as_nightly_cargo()
+        .with_stdout("")
```

Comment:
I'm not sure what this is trying to check, since output normally doesn't go to stdout.  If it is trying to check that it doesn't rebuild, I think checking stderr for `[FINISHED] [..]` would be good.

Reply:
Fixed.